src/comment.rs

```diff
@@ -47,39 +159,44 @@ pub fn rewrite_comment(orig: &str,
     if num_bare_lines > 0 && !config.normalize_comments() {
         return Some(orig.to_owned());
     }
-
     if !config.normalize_comments() && !config.wrap_comments() {
         return light_rewrite_comment(orig, shape.indent, config);
     }
 
+    let style = comment_style(orig, false);
+    let first_group = orig.lines()
+        .take_while(|l| style.to_pattern(l, orig, false))
+        .collect::<Vec<_>>()
+        .join("\n");
+    let rest = orig.lines()
+        .skip(first_group.lines().count())
+        .collect::<Vec<_>>()
+        .join("\n");
+
+    let first_group_str = try_opt!(rewrite_comment_inner(&first_group, block_style, shape, config));
+    if rest.is_empty() {
+        Some(first_group_str)
+    } else {
+        rewrite_comment(&rest, block_style, shape, config).map(|rest_str| {
```

Comment:
Rather than have `rewrite_comment` and `rewrite_comment_inner` where the latter recurses, could we have a `rewrite_comment` function which calls an `identify_comment` where `identify_comment` is recursive?